helper/plugin/grpc_provider.go

```diff
@@ -1291,3 +1321,52 @@ func normalizeNullValues(dst, src cty.Value, preferDst bool) cty.Value {
 
 	return dst
 }
+
+// validateConfigNulls checks a config value for unsupported nulls before
+// attempting to shim the value. While null values can mostly be ignored in the
+// configuration, since they're not supported in HCL1, the case where a null
+// appears in a list-like attribute (list, set, tuple) will present a nil value
+// to helper/schema which can panic. Return an error to the user in this case,
+// indicating the attribute with the null value.
+func validateConfigNulls(v cty.Value, path cty.Path) []*proto.Diagnostic {
+	var diags []*proto.Diagnostic
+	if v.IsNull() || !v.IsKnown() {
+		return diags
+	}
+
+	switch {
+	case v.Type().IsListType() || v.Type().IsSetType() || v.Type().IsTupleType():
+		it := v.ElementIterator()
+		for it.Next() {
+			kv, ev := it.Element()
+			if ev.IsNull() {
+				diags = append(diags, &proto.Diagnostic{
+					Severity:  proto.Diagnostic_ERROR,
+					Summary:   "null value found in list",
```

Comment:
Diagnostic summaries are normally written with a leading capital letter, like `"Null value in list"`.

Maybe should also add a `Detail` message here to say a similar thing in a more wordy way for those who find themselves asking "So?":

```go
Detail: "Null values are not allowed for this attribute value.",
```

Not particularly actionable, but at least follows the usual pattern for diagnostic messages. I considered making a specific suggestion for filtering out the nulls using language features, but decided against it because that would bake information about the current config language into the provider releases and that information may become stale in future versions, making the message misleading.

Reply:
ah yes, was following the Go style there, rather than the TF style